watertap/unit_models/translators/tests/test_translator_asm2d_adm1.py

```diff
@@ -634,3 +642,77 @@ def test_conservation(self, asmadm):
+    @pytest.mark.solver
+    def test_run_convergence_analysis_SF(self, asmadm):
+        spec = ParameterSweepSpecification()
+        spec.add_sampled_input(
+            "fs.unit.inlet.conc_mass_comp[0, S_F]", lower=0.001, upper=0.1
+        )
+        spec.set_sampling_method(UniformSampling)
+        spec.set_sample_size([4])
+
+        ca = IpoptConvergenceAnalysis(asmadm, input_specification=spec)
+
+        ca.run_convergence_analysis()
+
+        assert isinstance(ca.results, dict)
+        assert len(ca.results) == 4
+
+        assert ca.results[0]["success"]
+        assert ca.results[0]["results"]["iters"] == 3
+        assert ca.results[0]["results"]["iters_in_restoration"] == 0
+        assert ca.results[0]["results"]["iters_w_regularization"] == 0
+        assert ca.results[0]["results"]["numerical_issues"]
+
+        assert ca.results[1]["success"]
+        assert ca.results[1]["results"]["iters"] == 3
+        assert ca.results[1]["results"]["iters_in_restoration"] == 0
+        assert ca.results[1]["results"]["iters_w_regularization"] == 0
+        assert ca.results[1]["results"]["numerical_issues"]
+
+        assert ca.results[2]["success"]
+        assert ca.results[2]["results"]["iters"] == 3
+        assert ca.results[2]["results"]["iters_in_restoration"] == 0
+        assert ca.results[2]["results"]["iters_w_regularization"] == 0
+        assert ca.results[2]["results"]["numerical_issues"]
+
+        assert ca.results[3]["success"]
+        assert ca.results[3]["results"]["iters"] == 3
+        assert ca.results[3]["results"]["iters_in_restoration"] == 0
+        assert ca.results[3]["results"]["iters_w_regularization"] == 0
+        assert ca.results[3]["results"]["numerical_issues"]
+
+    @pytest.mark.integration
+    @pytest.mark.solver
+    def test_run_convergence_analysis_XS(self, asmadm):
+        spec = ParameterSweepSpecification()
+        spec.add_sampled_input(
+            "fs.unit.inlet.conc_mass_comp[0, X_S]", lower=1, upper=100
+        )
+        spec.set_sampling_method(UniformSampling)
+        spec.set_sample_size([4])
+
+        ca = IpoptConvergenceAnalysis(asmadm, input_specification=spec)
+
+        ca.run_convergence_analysis()
+
+        assert isinstance(ca.results, dict)
+        assert len(ca.results) == 4
+
+        assert ca.results[0]["success"]
+        assert ca.results[0]["results"]["iters"] == 3
+        assert ca.results[0]["results"]["iters_in_restoration"] == 0
+        assert ca.results[0]["results"]["iters_w_regularization"] == 0
+        assert ca.results[0]["results"]["numerical_issues"]
+
+        assert ca.results[1]["success"]
+        assert ca.results[1]["results"]["iters"] == 3
+        assert ca.results[1]["results"]["iters_in_restoration"] == 0
+        assert ca.results[1]["results"]["iters_w_regularization"] == 0
+        assert ca.results[1]["results"]["numerical_issues"]
+
+        assert not ca.results[2]["success"]
+
+        assert not ca.results[3]["success"]
```

Comment:
I like the inclusion of this test. However, we should be careful about how adding more of this test would affect testing times. Additionally, I believe that for this test (not in this case, as Marcus demonstrated in a previous comment) to effectively show whether convergence is improved, the tests should be conducted over the flowsheet rather than on the specific translators.

Reply:
There didn't appear to be any change in the flowsheet's model diagnostics, but I didn't do a parameter sweep, so I'll work on that soon (tomorrow?)